pkg/util/node/node.go

```diff
@@ -170,7 +170,7 @@ func SetNodeCondition(c clientset.Interface, node types.NodeName, condition v1.N
 	condition.LastHeartbeatTime = metav1.NewTime(time.Now())
 	patch, err := generatePatch(condition)
 	if err != nil {
-		return nil
+		return fmt.Errorf("failed to generate patch: %v", err)
```

Comment:
Have you analyzed all call-sites to convince yourself that this won't break something?  I took a quick look and it is not at all self-evident, IMO.  It needs to be though through.  Every place that calls this function now has a new early-termination case, and recursively so.

Is there a bug motivating this change? Or was it by inspection?

If you think this is right, please document here each of the call-paths and why you think this is OK.

Reply:
I think the difference is that, many callers actually retries if there's an error. In the case of marshaling error, retrying wouldn't help. The function could use better documentation for sure.